src/main/java/com/redhat/devtools/lsp4ij/server/definition/LanguageServerDefinitionListener.java

```diff
@@ -28,6 +29,7 @@ class LanguageServerAddedEvent {
 
         public LanguageServerAddedEvent(@NotNull Collection<LanguageServerDefinition> serverDefinitions) {
             this.serverDefinitions = serverDefinitions;
+            LanguageServiceAccessor.checkCurrentlyOpenFiles(serverDefinitions);

```

Comment:
This check must not be done when add event is created.

The language service accessor must add a l'istener and track the add event